catalog/spatial/geocoding/spatial-geocoding-offline-catalog/src/main/java/org/codice/ddf/spatial/geocoding/query/GazetteerQueryCatalog.java

```diff
@@ -144,6 +149,40 @@ public GazetteerQueryCatalog(CatalogFramework catalogFramework, FilterBuilder fi
         .collect(Collectors.toList());
   }
 
+  @Override
+  public List<String> getSuggestedNames(String queryString, int maxResults)
+      throws GeoEntryQueryException {
+    Map<String, Serializable> suggestProps = new HashMap<>();
+    suggestProps.put(SUGGESTION_REQUEST_KEY, queryString);
+
+    Query suggestionQuery = new QueryImpl(filterBuilder.attribute(Core.TITLE).text(queryString));
+
+    QueryRequest suggestionRequest = new QueryRequestImpl(suggestionQuery, suggestProps);
+
+    try {
+      QueryResponse suggestionResponse = catalogFramework.query(suggestionRequest);
+
+      Map<String, List<Suggestion>> suggestionMap =
+          (Map<String, List<Suggestion>>)
+              suggestionResponse.getPropertyValue(SUGGESTION_RESULT_KEY);
+      if (suggestionMap != null) {
+        return suggestionMap
+            .values()
+            .stream()
+            .flatMap(List::stream)
+            .limit(maxResults)
+            .map(Suggestion::getTerm)
+            .collect(Collectors.toList());
+      }
+
+    } catch (SourceUnavailableException | FederationException | UnsupportedQueryException e) {
+      LOGGER.debug("Suggestion query failed", e);
```

Comment:
I think log n' throw is generally considered a code smell; i think the exception will be enough in this case.